opengrok-indexer/src/main/java/org/opengrok/indexer/history/History.java

```diff
@@ -123,4 +123,27 @@ public boolean isRenamed(String file) {
     public Set<String> getRenamedFiles() {
         return renamedFiles;
     }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        History that = (History) o;
+        return Objects.equals(this.getHistoryEntries(), that.getHistoryEntries()) &&
+                Objects.equals(this.getRenamedFiles(), that.getRenamedFiles());
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(getHistoryEntries().hashCode(), getRenamedFiles().hashCode());
```

Comment:
You are not supposed to call `hashCode()` on the objects, the implementation should handle that :) 
```suggestion
        return Objects.hash(getHistoryEntries(), getRenamedFiles());
```